language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/EventQueryClientImplTest.scala

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.rxjava.grpc
+
+import com.daml.ledger.api.auth.{AuthService, AuthServiceWildcard}
+import com.daml.ledger.api.v1.ValueOuterClass
+import com.daml.ledger.rxjava.grpc.helpers.TestConfiguration
+import org.scalacheck.Arbitrary
+
+import com.daml.ledger.api.v1.event_query_service.{
+  GetEventsByContractIdResponse,
+  GetEventsByContractKeyResponse,
+}
+import com.daml.ledger.javaapi.data._
+import com.daml.ledger.rxjava._
+import com.daml.ledger.rxjava.grpc.helpers.{DataLayerHelpers, LedgerServices}
+import org.scalatest.OptionValues
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import java.util.Optional
+import java.util.concurrent.TimeUnit
+import scala.concurrent.Future
+import scala.jdk.CollectionConverters._
+
+class EventQueryClientImplTest
+    extends AnyFlatSpec
+    with Matchers
+    with AuthMatchers
+    with OptionValues
+    with DataLayerHelpers {
+
+  val ledgerServices = new LedgerServices("event-query-service-ledger")
+
+  private def withEventQueryClient(authService: AuthService = AuthServiceWildcard) = {
+    ledgerServices.withEventQueryClient(
+      Future.successful(GetEventsByContractIdResponse.defaultInstance),
+      Future.successful(GetEventsByContractKeyResponse.defaultInstance),
+      authService,
+    ) _
+  }
+
+  private val contractKey = Value.fromProto(
+    Arbitrary.arbBool.arbitrary
+      .map(ValueOuterClass.Value.newBuilder().setBool(_).build())
+      .sample
+      .get
+  )
```

Comment:
Considering it just represents a bool, I don't think there's much need for the fuzzing.

```suggestion
  private val contractKey = Bool.TRUE
```